### PR TITLE
feat(questions): display update date on questions and share component

### LINKED ISSUE #2960: [QuestionPage] Created/last updated date

### PR DESCRIPTION
**Pull request by oktomus**
Question, Research and How-To modules now use the same component to display the author, create and update dates.

PR Checklist

- [x] - Commit [messages are descriptive](https://github.com/ONEARMY/community-platform/blob/master/CONTRIBUTING.md#--commit-style-guide), it will be used in our [Release Notes](https://github.com/ONEARMY/community-platform/releases/)

PR Type

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Developer experience (improves developer workflows for contributing to the project)

## Description

- Show update date on questions
- Introduce a new react component to display the author, created and updated dates

## Git Issues

Closes #2960

---

## What happens next?

Thanks for the contribution! We try to make sure all PRs are reviewed ahead of our monthly maintainers call (first Monday of the month)

If the PR is working as intended it'll be merged and included in the next platform release, if not changes will be requested and re-reviewed once updated.

If you need more immediate feedback you can try reaching out on Discord in the [Community Platform `development` channel](https://discord.com/channels/586676777334865928/938781727017558018).

### CODE REVIEW

**Comment by oktomus** [comment]
Working on the failing tests.

**Comment by oktomus** [comment]
Done. Should we write tests for questions using cypress like [here](https://github.com/ONEARMY/community-platform/blob/dc6846d250bd5d529bb000b45afaff9e6bda3f0d/packages/cypress/src/integration/howto/read.spec.ts)?

**Comment by thisislawatts** [comment]
Good question @oktomus, I am actually wondering if we can achieve enough confidence with lower level (faster) tests and minimise our Cypress tests to a few key flows. What do you think?

File: src/pages/common/ModeratedContentTags/ModeratedContentTags.tsx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+interface ModeratedContentTagsProps {
+  userName: string
+  countryCode: string | undefined
+  created: string | number | Date
+  modified: string | number | Date
+  action?: string
+}
+
+export const ModeratedContentTags = ({
```
Comment (thisislawatts) [inline src/pages/common/ModeratedContentTags/ModeratedContentTags.tsx:13]: **nitpick:** The phrase `ModeratedContentTags` doesn't have much meaning to me. Could we call it something like `ArticleAuthorTimestamp` or `ContentAuthorTimestamp`?

For broader context here there are some high level plans about migrating Research/Howto/Question to use a single underlying data structure, this new entity would be known as an Article with various behaviours which then present as either Research, Howto or Question pages.

Comment (thisislawatts) [reply src/pages/common/ModeratedContentTags/ModeratedContentTags.tsx:13]: **praise:** Thanks for taking the time to think about and refactor these into a single reusable component :sparkles:

Comment (oktomus) [reply src/pages/common/ModeratedContentTags/ModeratedContentTags.tsx:13]: Let's go with `ContentAuthorTimestamp`, I like it :)

**Comment by oktomus** [comment]
> Good question @oktomus, I am actually wondering if we can achieve enough confidence with lower level (faster) tests and minimise our Cypress tests to a few key flows. What do you think?

I don't have experience with web tests and web specs. 

Looking at Cypress tests, they look easy to implement so I would keep them. 

I feel like I can already implement Cypress tests, but not the other kind of tests I saw on the project.

**Review by thisislawatts — APPROVED** [review]

**Comment by thisislawatts** [comment]
@all-contributors add @oktomus for code

**Pull request merged**